yt_dlp/extractor/youmaker.py

```diff
@@ -0,0 +1,430 @@
+                yield self.url_result(
+                    f'{self._base_url}/video/{video_uid}',
+                    ie=self.ie_key(),
+                    video_id=video_uid,
+                    video_title=title)
+
+        _ = self._categories  # preload categories
+        return OnDemandPagedList(fetch_page, page_size)
+
+    def _paged_channel_entries(self, uid, page_size=REQUEST_LIMIT):
+        def fetch_page(page_number):
+            offset = page_number * page_size
+            info = self._call_api(
+                uid,
+                path=f'video/channel/{uid}',
+                what=f'channel entries {offset + 1}-{offset + page_size}',
+                query={'offset': offset, 'limit': page_size})
+            if not isinstance(info, list):
+                raise ExtractorError('Unexpected channel entries', uid, expected=False)
+
+            for item in info:
+                video_uid, title = itemgetter('video_uid', 'title')(item)
+                self._cache[video_uid] = item
+                yield self.url_result(
+                    f'{self._base_url}/video/{video_uid}',
+                    ie=self.ie_key(),
+                    video_id=video_uid,
+                    video_title=title)
+
+        _ = self._categories  # preload categories
+        return OnDemandPagedList(fetch_page, page_size)
+
+    def _playlist_entries_by_id(self, uid):
+        _ = self._categories  # preload categories
+        info = self._call_api(uid, f'playlist/{uid}', what='playlist metadata')
+        return self.playlist_result(
+            self._paged_playlist_entries(info['playlist_uid']),
+            playlist_id=info['playlist_uid'],
+            playlist_title=info.get('name'),
+            playlist_description=None)
+
+    def _channel_entries_by_id(self, uid):
+        _ = self._categories  # preload categories
+        info = self._call_api(
+            uid,
+            path=f'video/channel/metadata/{uid}',
+            what='channel metadata')
+        return self.playlist_result(
+            self._paged_channel_entries(info['channel_uid']),
+            playlist_id=info['channel_uid'],
+            playlist_title=info.get('name'),
+            playlist_description=info.get('description'))
+
+    def _real_extract(self, url):
+        parsed_url = ParsedURL(url)
+        self._protocol = parsed_url.scheme
+
+        dispatch = (
+            (r'/(?:v|video|embed)/(?P<uid>[a-zA-z0-9-]+)', self._video_entry_by_id),
+            (r'(/channel/[a-zA-z0-9-]+)?/playlists?/(?P<uid>[a-zA-z0-9-]+)',
+             self._playlist_entries_by_id),
+            (r'/channel/(?P<uid>[a-zA-z0-9-]+)/?$', self._channel_entries_by_id),
+        )
+
+        for regex, func in dispatch:
+            match = re.match(regex, parsed_url.path)
+            if not match:
+                continue
+            return func(**match.groupdict())
+        else:
+            raise ExtractorError(f'unsupported {self.ie_key()} url', expected=True)
```

Comment:
This seems exactly like what having different extractors with different `_VALID_URLS` automatically achieve. I dont see any reason why you are simulating the same behaviour in this way and making it unnecessarily complex